src/test/run-make-fulldeps/sanitizer-leak/leak.rs

```diff
@@ -1,6 +1,12 @@
+#![feature(test)]
+
+use std::hint::black_box;
 use std::mem;
 
 fn main() {
-    let xs = vec![1, 2, 3, 4];
-    mem::forget(xs);
+    for _ in 0..10 {
+        let xs = vec![1, 2, 3];
+        let xs = black_box(xs);
```

Comment:
This looks like it is using `black_box` for correctness purposes, contrary to https://github.com/rust-lang/rfcs/pull/2360? cc @eddyb

Reply:
How's this about correctness? Looks similar to benchmarks for me, where we want to stere the compiler towards generating the "right" assembly as that's what we want to test.

In particular, the program is still UB-free even if `black_box(xs)` is replaced by `xs`.

It'd be good to see a comment here, though.

Reply:
Now with a comment.